source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,379 @@ Documents
+read our page on :manual:`Documents </core/document>` in the MongoDB manual.
+
+The Java driver and BSON library include the following document classes that
+help you access and manipulate the BSON data in documents:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 10 20 20
+
+   * - Name
+     - Package
+     - Implements Map
+     - Recommended Usage
+
+   * - ``Document``
+     - ``org.bson``
+     - Yes, implements ``Map<String, Object>``
+     - When you want a flexible and concise data representation.
+
+   * - ``BsonDocument``
+     - ``org.bson``
+     - Yes, implements ``Map<String, BsonValue>``
+     - When you need a type-safe API.
+
+   * - ``BasicDBObject``
+     - ``com.mongodb``
+     - No
+     - When you are migrating your application from a legacy driver version.
+
+While you can use any of the document classes in your application, we
+recommend that you use the ``Document`` class since it can concisely
+represent dynamically structured documents of any complexity. It
+implements the ``Map<String, Object>`` interface which enables it to use
+loosely-typed values.
 
 Document
 --------
 
-- Description
-- Chart that shows mapping between BSON and Java types (similar to the one here https://mongodb.github.io/mongo-java-driver/4.1/bson/documents/)
-- Code example of declaring one and appending fields
-- Code example of get() for various types
-- Link to Fundamentals > CRUD for more information
-- Link to API documentation
+The ``Document`` class offers a flexible representation of a BSON document.
+You can access and manipulate fields using Java types with this class. See
+the following table for mappings between frequently-used BSON and Java types:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - BSON type
+     - Java type
+   * - Array
+     - ``java.util.List``
+   * - Binary
+     - ``org.bson.types.Binary``
+   * - Boolean
+     - ``java.lang.Boolean``
+   * - Date
+     - ``java.util.Date``
+   * - Document
+     - ``org.bson.Document``
+   * - Double
+     - ``java.lang.Double``
+   * - Int32
+     - ``java.lang.Integer``
+   * - Int64
+     - ``java.lang.Long``
+   * - Null
+     - ``null``
+   * - ObjectId
+     - ``org.bson.types.ObjectId``
+   * - String
+     - ``java.lang.String``
+
+The mapping table above shows the default mapping when working with the
+``Document`` class. You can customize the type mapping by specifying a
+custom codec. For more information on customizing mapped types, see our guide
+on using :doc:`Codecs </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate and build a sample
+``Document`` instance representing a document containing several
+different field types:
+
+.. code-block:: java
+
+   Document author = new Document("_id", new ObjectId())
+       .append("name", "Gabriel García Márquez")
+       .append("lastUpdated", new Date(120, 2, 26))
```

Comment:
That's a strange date... The year 120?  

Also, that constructor is deprecated.  Horribly, the way to do this in Java is:

    Date.from(LocalDate.of(2000, 1, 1).atStartOfDay(ZoneId.systemDefault()).toInstant())

But since you're just modeling a lastUpdated field, why not just `new Date())`?

Reply:
Thanks, haven't kept up with Java 8+ dates. I'll change the field name and purpose since I want to show an example of setting a specific date.